tests/test_permissions_synchronizer.py

```diff
@@ -667,6 +667,97 @@ def test_webhooks_invalid_service(self):
+        """
+        Tests the permissions synchronization of resources that use a regex to extract a display_name in the config.
+        """
+        self.data["sync_permissions"] = {
+            "user_workspace": {
+                "services": {
+                    ServiceTHREDDS.service_type: {
+                        "Thredds_file_src": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"name": "private", "type": Directory.resource_type_name},
+                            {"name": "directory1", "type": Directory.resource_type_name},
+                            {"field": "resource_display_name",
+                             "regex": r"[\w]+:[\w\/.-]+",
+                             "type": File.resource_type_name}],
+                        "Thredds_file_target": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"regex": r"(?<=catalog:)[\w\/]*\/?(?=\/)", "type": Directory.resource_type_name},
+                            {"regex": r"[^\/]+$", "type": File.resource_type_name}],
+                        "Thredds_dir_src": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"name": "private", "type": Directory.resource_type_name},
+                            {"field": "resource_display_name", "regex": r"[\w]+:[\w\/]+",
+                             "type": Directory.resource_type_name}],
+                        "Thredds_dir_target": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"regex": r"(?<=:)[\w\/]+", "type": Directory.resource_type_name}]}},
+                "permissions_mapping": [
+                    f"Thredds_file_src : {Permission.READ.value} -> "
+                    f"Thredds_file_target : {Permission.READ.value}",
+                    f"Thredds_dir_src : {Permission.READ.value} -> "
+                    f"Thredds_dir_target : {Permission.READ.value}",
+                ]
+            }
+        }
+        with open(self.cfg_file.name, mode="w", encoding="utf-8") as f:
+            f.write(yaml.safe_dump(self.data))
+        app = utils.get_test_app(settings={"cowbird.config_path": self.cfg_file.name})
+        # Recreate new magpie handler instance with new config
+        HandlerFactory().create_handler("Magpie")
+
+        with contextlib.ExitStack() as stack:
+            stack.enter_context(mock.patch("cowbird.handlers.impl.thredds.Thredds",
+                                           side_effect=utils.MockAnyHandler))
+
+            # Create test resources
+            dir_src_res_id = self.magpie.create_resource("private", Directory.resource_type_name, self.test_service_id)
+            parent_res_id = self.magpie.create_resource("directory", Directory.resource_type_name, dir_src_res_id,
```

Comment:
Why is the resource named `directory`here, but is named `directory` in the config above?

Reply:
Since the test is using one service to test a mapping between two different path. I used `catalog/directoy/file.nc ` and `catalog/private/directoy/file.nc`.

Reply:
Sorry for the confusion, I missed the `1` in the `directory1` I was meant to reference in the config above. Was only wondering about the difference in the names `directory` and `directory1`.

Reply:
oh okay I see, it's a typo will fix